pkg/ibm_plugin/server/plugin_integration_test.go

```diff
@@ -0,0 +1,406 @@
+//go:build ibm
+
+/*
+Copyright 2023 The Paraglider Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ibm
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+	"github.com/google/uuid"
+	"github.com/paraglider-project/paraglider/pkg/azure"
+	ibmCommon "github.com/paraglider-project/paraglider/pkg/ibm_plugin"
+	sdk "github.com/paraglider-project/paraglider/pkg/ibm_plugin/sdk"
+	"github.com/paraglider-project/paraglider/pkg/kvstore"
+	"github.com/paraglider-project/paraglider/pkg/orchestrator"
+	"github.com/paraglider-project/paraglider/pkg/orchestrator/config"
+	"github.com/paraglider-project/paraglider/pkg/paragliderpb"
+	tagging "github.com/paraglider-project/paraglider/pkg/tag_service"
+	utils "github.com/paraglider-project/paraglider/pkg/utils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var testDeployment, resourceGroupID string
+
+// NOTE: if user doesn't have resource group privileges set DoesHaveResourceGroupPrivileges to false
+const DoesHaveResourceGroupPrivileges = false
+
+// NOTE: if user doesn't have resource group privileges set azureResourceGroupName to match an existing resource group
+var azureResourceGroupName string = "challenge-1377"
+
+func TestMain(m *testing.M) {
```

Comment:
What is this function for?

Reply:
It allows me to perform the necessary setup before tests execution.